### PR TITLE
Show custom or default module name on the second tile

### PR DESCRIPTION
**Pull request by kakhahmed**
I wanted to show the module names as we are doing for AGIPD, LPD, and DSSC. Looking at the other implementations I realized there is a style of showing the module name on one of the tiles.  I picked the second one and I didn't have a strong opinion against the tile names so I left them and only removed the one with the module name.

I have added an option for JUNGFRAU only to replace the module names with custom ones. I had a case where I would like to show the name of the module and the name of the PDU.

A screenshot after inspecting a `JUNGFRAUGeometry` object with the default module names.

Screenshot from 2022-09-21 12-46-58

And here are with the custom module names

Screenshot from 2022-09-21 12-46-52

@takluyver

### CODE REVIEW

**Comment by kakhahmed** [comment]
Screenshot from 2022-09-28 17-10-15

Screenshot from 2022-09-28 17-10-09

Screenshot from 2022-09-28 17-10-03


These are the new assembelled preview with the latest changes. Don't you feel it looking a bit weird for the 1 and 2 modules? Is there a way were I can show the module name in the center of the module itself and not within a selected tile?

File: extra_geom/detectors.py [hunk 1]
```diff
@@ -1644,14 +1644,24 @@ def inspect(self, axis_units='px', frontview=True):
...
-            tiles = self.modules[m]
+        for i, m in enumerate(self.modules):
+            cx, cy, _ = m[1].centre() * scale
+            module_text = module_names[i] if module_names else f'JNGFR{i+1:02d}'
+            ax.text(cx, cy, module_text, fontweight='bold',
+                    size='x-large',
+                    verticalalignment='center',
+                    horizontalalignment='center')
+
+            tiles = self.modules[i]
```
[the remark this answers is not in the document]
Comment (kakhahmed) [reply extra_geom/detectors.py]: Thank you, the suggestion is added.

**Comment by kakhahmed** [comment]
I have updated the documentation as well.

**Comment by kakhahmed** [comment]
> I'd like to keep the numbering on the first tile - maybe we could move the module name one square over. We could also reduce the labelling on the individual tiles, e.g. just the first and last, and remove the `M0` parts (which don't match up with the module names). If there's any convention for how to refer to the tiles within JUNGFRAU modules (e.g. maybe they're T1 - T8?), we should follow that.

Thanks for the suggestion. I have confirmed with Marco and he prefers having only Ax as well without the Mx part. I have also left the first and last ASIC labels.

**Comment by takluyver** [comment]
Thanks!

I agree it looks a bit weird for 1/2 modules, but I don't think it's a big deal. Thinking about it, though, maybe we should put A4 back in, to make it clear that the numbers count across the rows, not along each column.

There isn't a built-in way to find the module centre, though you could calculate it from the tile centres or corners. But for custom names, that would give less available width for custom names before you're going onto the A0/A4/A7 tiles, so I might leave it as is.

**Comment by kakhahmed** [comment]
> I agree it looks a bit weird for 1/2 modules, but I don't think it's a big deal. Thinking about it, though, maybe we should put A4 back in, to make it clear that the numbers count across the rows, not along each column.

Done, A4 is displayed as well.

> There isn't a built-in way to find the module centre, though you could calculate it from the tile centres or corners. But for custom names, that would give less available width for custom names before you're going onto the A0/A4/A7 tiles, so I might leave it as is.


Fine by me. I think the MR is finished from my side.


Thank you for the nice discussion and review.

**Comment by takluyver** [comment]
Thanks @kakhahmed , this LGTM

**Pull request merged**